tests/python/CPUProcessorTest.py

```diff
@@ -0,0 +1,629 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+
+import logging
+import unittest
+
+logger = logging.getLogger(__name__)
+
+try:
+    import numpy as np
+except ImportError:
+    logger.warning(
+        "NumPy could not be imported. "
+        "Test case will lack significant coverage!"
+    )
+    np = None
+
+import PyOpenColorIO as OCIO
+
+
+class CPUProcessorTest(unittest.TestCase):
+    FLOAT_DELTA = 1e+5
+    UINT_DELTA = 1
+
+    @classmethod
+    def setUpClass(cls):
+        # ---------------------------------------------------------------------
+        # OCIO objects
+        # ---------------------------------------------------------------------
+        cls.config = OCIO.Config()
+        cls.proc_fwd = cls.config.getProcessor(
+            OCIO.MatrixTransform.Scale([0.5, 0.5, 0.5, 1.0])
+        )
+        cls.proc_inv = cls.config.getProcessor(
+            OCIO.MatrixTransform.Scale([2.0, 2.0, 2.0, 1.0])
+        )
+
+        # BIT_DEPTH_F32
+        cls.default_cpu_proc_fwd = cls.proc_fwd.getDefaultCPUProcessor()
+        cls.default_cpu_proc_inv = cls.proc_inv.getDefaultCPUProcessor()
+
+        # BIT_DEPTH_F16
+        cls.half_cpu_proc_fwd = cls.proc_fwd.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_F16, 
+            OCIO.BIT_DEPTH_F16, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+        cls.half_cpu_proc_inv = cls.proc_inv.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_F16, 
+            OCIO.BIT_DEPTH_F16, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+
+        # BIT_DEPTH_UINT16
+        cls.uint16_cpu_proc_fwd = cls.proc_fwd.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_UINT16, 
+            OCIO.BIT_DEPTH_UINT16, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+        cls.uint16_cpu_proc_inv = cls.proc_inv.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_UINT16, 
+            OCIO.BIT_DEPTH_UINT16, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+
+        # BIT_DEPTH_UINT8
+        cls.uint8_cpu_proc_fwd = cls.proc_fwd.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_UINT8, 
+            OCIO.BIT_DEPTH_UINT8, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+        cls.uint8_cpu_proc_inv = cls.proc_inv.getOptimizedCPUProcessor(
+            OCIO.BIT_DEPTH_UINT8, 
+            OCIO.BIT_DEPTH_UINT8, 
+            OCIO.OPTIMIZATION_DEFAULT
+        )
+
+        # ---------------------------------------------------------------------
+        # Test image data
+        # ---------------------------------------------------------------------
+        if not np:
+            # NumPy not found. Only test lists.
+            log_range = [float(2**i) for i in range(-15, 15)]
+
+            cls.float_rgb_list = (
+                [-65504.0] + 
+                list(reversed(log_range)) + 
```

Comment:
Did you meant `[-x for x in reversed(log_range)]` ?

Sorry for the late comment on this great PR.

Reply:
Yes, good catch yet again! Will fix on my next pass.